db/agendadb/db.go

```diff
@@ -0,0 +1,227 @@
+	Id             string           `json:"id" storm:"id"`
+	Description    string           `json:"description"`
+	Mask           uint16           `json:"mask"`
+	StartTime      uint64           `json:"starttime" storm:"index"`
+	ExpireTime     uint64           `json:"expiretime" storm:"index"`
+	Status         string           `json:"status" storm:"index"`
+	QuorumProgress float64          `json:"quorumprogress" storm:"index"`
+	Choices        []dcrjson.Choice `json:"choices"`
+	VoteVersion    uint32           `json:"voteversion"`
+}
+
+// ChoiceLabeled embeds dcrjson.Choice along with the AgendaID for the choice,
+// and a string array suitable for use as a primary key. The AgendaID is tagged
+// as an index for quick lookups based on the agenda.
+type ChoiceLabeled struct {
+	AgendaChoice   [2]string `storm:"id"`
+	AgendaID       string    `json:"agendaid" storm:"index"`
+	dcrjson.Choice `storm:"inline"`
+}
+
+// GetVoteAgendasForVersion is used in getting the agendas using the vote versions
+func GetVoteAgendasForVersion(ver int64, client *rpcclient.Client) (agendas []AgendaTagged) {
+	voteInfo, err := client.GetVoteInfo(uint32(ver))
+	if err != nil {
+		log.Errorf("Fetching Agendas by vote version failed: %v", err)
+	} else {
+		for i := range voteInfo.Agendas {
+			v := &voteInfo.Agendas[i]
+			a := AgendaTagged{
+				Id:             v.Id,
+				Description:    v.Description,
+				Mask:           v.Mask,
+				StartTime:      v.StartTime,
+				ExpireTime:     v.ExpireTime,
+				Status:         v.Status,
+				QuorumProgress: v.QuorumProgress,
+				Choices:        v.Choices,
+				VoteVersion:    voteInfo.VoteVersion,
+			}
+			agendas = append(agendas, a)
+		}
+	}
+	return
+}
+
+// CheckAvailabiltyOfVersionAgendas checks for the availabily of agendas in the
+// saved db by version input
+func (db *AgendaDB) CheckAvailabiltyOfVersionAgendas(version int64) bool {
+	if db == nil || db.sdb == nil {
+		return false
+	}
+
+	agenda := make([]AgendaTagged, 0)
+	err := db.sdb.Find("VoteVersion", uint32(version), &agenda)
+	if len(agenda) == 0 || err != nil {
+		return false
+	}
+
+	return true
+}
+
+// Updatedb used when needed to keep the saved db upto date
+func (db *AgendaDB) Updatedb(voteVersion int64, client *rpcclient.Client) {
+	var agendas []AgendaTagged
+	for GetVoteAgendasForVersion(voteVersion, client) != nil {
+		taggedAgendas := GetVoteAgendasForVersion(voteVersion, client)
+		if len(taggedAgendas) > 0 {
+			agendas = append(agendas, taggedAgendas...)
+			voteVersion++
+		}
+	}
+	for i := range agendas {
+		err := db.StoreAgenda(&agendas[i])
+		if err != nil {
+			log.Errorf("Agenda not saved: %v \n", err)
+		}
+	}
+}
+
+// CheckForUpdates checks for update at the start of the process and will
+// proceed to update when neccessary.
+func CheckForUpdates(client *rpcclient.Client) error {
+	adb, err := Open(dbName)
+	if err != nil {
+		log.Errorf("Failed to open new DB: %v", err)
+		return nil
+	}
+	var voteVersion int64 = 4
```

Comment:
Hard coded values ("magic numbers") are bad news in general, and especially bad if there's no comments to explain what it's doing there.  e.g. It starts at 4 because...

Reply:
This was @McEdward Implementation. I will get more information on this add a comment to it.